src/test/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2Test.java

```diff
@@ -342,6 +352,46 @@ public void testUploadIrbDocumentWithPreviousIrbDocument() throws Exception {
     assertEquals(200, response.getStatus());
   }
 
+  private Pair<InputStream, FormDataContentDisposition> mockFormDataMultiPart(String fileName)
+      throws IOException {
+    String name = FilenameUtils.removeExtension(fileName);
+    String ext = FilenameUtils.getExtension(fileName);
+
+    InputStream inputStream = IOUtils.toInputStream(name, Charset.defaultCharset());
+    FormDataContentDisposition formData = mock(FormDataContentDisposition.class);
+    when(formData.getFileName()).thenReturn(fileName);
+    when(formData.getType()).thenReturn(ext);
+    when(formData.getSize()).thenReturn(1L);
+    when(formData.getName()).thenReturn(name);
+
+    return Pair.of(inputStream, formData);
+  }
+
+  @Test
+  public void testPostProgressReportCollabAndEthicsFiles() throws IOException {
+    when(userService.findUserByEmail(user.getEmail())).thenReturn(user);
+    DataAccessRequest parentDar = generateDataAccessRequest();
+    when(dataAccessRequestService.findByReferenceId(any())).thenReturn(parentDar);
+    DataAccessRequest childDar = generateDataAccessRequest();
+    when(dataAccessRequestService.createDataAccessRequest(any(), any())).thenReturn(childDar);
+    Pair<InputStream, FormDataContentDisposition> collabFile = mockFormDataMultiPart("collab.txt");
+    Pair<InputStream, FormDataContentDisposition> ethicsFile = mockFormDataMultiPart("ethics.txt");
+    Dataset dataset = mock(Dataset.class);
+    when(datasetService.findDatasetById(any())).thenReturn(dataset);
+    DataUse dataUseCollabAndEthics = new DataUseBuilder()
+        .setCollaboratorRequired(true)
+        .setEthicsApprovalRequired(true).build();
+    when(dataset.getDataUse()).thenReturn(dataUseCollabAndEthics);
+    when(gcsService.storeDocument(any(), any(), any())).thenReturn(BlobId.of("bucket", "name"));
+    when(gcsService.deleteDocument(any())).thenReturn(true);
+    when(dataAccessRequestService.updateByReferenceId(any(), any())).thenReturn(childDar);
+    initResource();
+
+    Response response = resource.postProgressReport(authUser, "", "", collabFile.getLeft(),
+        collabFile.getRight(), ethicsFile.getLeft(), ethicsFile.getRight());
+    assertEquals(200, response.getStatus());
+  }
```

Comment:
Some additional test case ideas
* Null dataset.getDataUse()
* Null dataUse.getCollaboratorRequired()
* Null dataUse.getEthicsApprovalRequired()
* Non-create user trying to create a PR
* Parent DAR not found
* Requested datasets not found